### DISCUSSION TITLE
Preloading screen inside nested stack navigator

### DISCUSSION THREAD

**Discussion by quiloos39** [react-navigation/react-navigation#9903] [Q&A] [answered]
Hello everyone i am having difficulty solving following problem 

![image](https://user-images.githubusercontent.com/10371011/132051904-5b2a06bd-b241-422f-a19b-e1684c5d0862.png)

I have navigation schema above and i want to navigate from A to C but before navigating to C i want to make sure that B is in stack history of stack navigator. 

Because for example if i am making anime app and inside of the app i have following navigation flow: anime banner (A) -> anime category (B) -> anime info (C)

If i directly navigate from A to C. I will loose category screen (B) and i will not able to go back to it anymore. Anime fans will be sad and i will be sad too. Is there way to preload screen B while app is starting ?

thanks ^^

**Comment by raajnadar** [comment, marked as answer]
Use the `navigation.reset` or update the nav history https://reactnavigation.org/docs/navigation-state/

**Reply by quiloos39** [reply]
I found alternative solution where i disabled lazy loading of tab navigation and in stack navigation set initial screen as B but thanks for help anyway

**Reply by raajnadar** [reply]
Please add your solution in detail & mark this discussion as solved.
